### ISSUE TITLE
documentation updates

### ISSUE THREAD

**Issue by bbbart** [dairiki/lektor-index-pages#1]
Hi,

first and foremost: thank you for this amazing lektor plugin. It's saving me a lot of time with my latest website project.

After using it for a while, I would like to suggest two documentation updates:
 * something that I figured out by simply trying it: it's trivially possible with this plugin to also create indexes with a key based on values of the type `strings`. I prefer using `strings` for tags and categories over lists prepared a hidden Lektor content as it's more flexible.
 * something that took me some time to figure out was how to assign simple string values to index fields. I ended up with working quoting eventually: `title = '"this is my title"'

Perhaps it would be nice to put this somewhere in the otherwise good documentation.

**Comment by dairiki** [comment]
Thank you for the thanks!

> * something that I figured out by simply trying it: it's trivially possible with this plugin to also create indexes with a key based on values of the type `strings`. I prefer using `strings` for tags and categories over lists prepared a hidden Lektor content as it's more flexible.

I myself have done exactly that for a "tags" index on a blog.

> * something that took me some time to figure out was how to assign simple string values to index fields. I ended up with working quoting eventually: `title = '"this is my title"'

I hadn't really thought that constant values for the fields was a useful pattern.  But, if it works for you, there’s certainly nothing wrong with it.  You have figured out that whatever value you set for a field in the `configs/index-pages.ini` file is evaluated as a Jinja expression to get the final field value — thus the need for quoting constant strings.

> Perhaps it would be nice to put this somewhere in the otherwise good documentation.

I will work on that.  (Pull requests are always welcome, too.)

**Comment by bbbart** [comment]
Ah, yes, I'm also using strings for tags, but was using the word categories in my original message to refer to the existing documentation. :-)

About the constant value pattern, let me share what I ended up with in my config file:

    [tags.fields]
    tag = this.key
    sidebar = true
    title = '"Index for tag '" ~ this.tag ~ "'" ~ (" (page " ~ this.pagination.page ~ ")" if this.pagination.page > 1)'

Some of this could have gone into the template file as well, but I am using the same template file for different indices, so it made more sense to me to put it in the configuration file.

Let me see if I can find some time for a PR. I have contributed documentation before and actually found it enjoyable to do.

Thanks for the quick response!

**Comment by bbbart** [comment]
pull request made: #2

**Comment by dairiki** [comment]
Thank you for the example.  I understand the issue with the fields better now.

Apparently, [inifile][], which is what Lektor uses to parse `.ini` files strips the outer quote marks.

The quotes are necessary if one wants to include escapes like `\n` within the value. Without the quotes that would be interpreted as a literal backslash followed by an "n".  With the quotes that gets interpreted by `inifile` as a newline character.  The quotes are also necessary if one wants to include leading or trailing white-space in the value.

From reading the [source code][], it appears that inifile’s handling of quoted strings is a bit wonky (as you noticed), in that it only recognizes as quoted values which begin and end with a (the same) quote character (`"` or `'`).

For example, inifile thinks this value is quoted, so it will strip the outer quotes:
```
title = "Index for tag '" ~ this.tag ~ "'"
```
resulting in title being set to `Index for tag '" ~ this.tag ~ "'`, which will result in a syntax error when this gets Jinja-evaluated.

On the other hand, this would not be treated as a quoted value by inifile, and so should work:
```
title = 'Index for tag \'' ~ this.tag ~ "'"
```
(The initial quote does not match the trailing quote, so `inifile` does not interpret this as a quoted valued.)

It is confusing.  I'm not sure what can be done about it.  Most of the issue seems to have to do with how `inifile` handles quoting — and that's not easy to change without breaking existing code.

It would be good to add a note about this to the documentation.  If you'd like to add something about this to your PR (#2), that would be wonderful.  Otherwise, I will work on it.

[inifile]: <https://github.com/mitsuhiko/python-inifile>
[source code]: <https://github.com/mitsuhiko/python-inifile/blob/ef31a6313e40ba9e649b4800c33d8366c2cf6813/inifile.py#L211>

**Comment by dairiki** [comment]
Closed via #2.  Thank you!
